From: "B. Stack" <bgstack15@gmail.com>
Date: Mon, 31 Aug 2020 00:00:00 +0000
Subject: Build with GTK3 instead of GTK2

Source: Original research
Forwarded: not-needed

- add the LDFLAGS for gtk+-3.0, otherwise, the build would fail with:

/usr/bin/ld: /tmp/cc6adrRa.ltrans115.ltrans.o: undefined reference to symbol 'gtk_css_provider_load_from_path'
/usr/bin/ld: /lib/x86_64-linux-gnu/libgtk-3.so.0: error adding symbols: DSO missing from command line

"::gtk_css_provider_load_from_path" is called in
  FreeFileSync/Source/application.cpp
  FreeFileSync/Source/RealTimeSync/application.cpp
---
 FreeFileSync/Source/Makefile              | 1 +
 FreeFileSync/Source/RealTimeSync/Makefile | 1 +
 2 files changed, 2 insertions(+)

diff --git a/FreeFileSync/Source/Makefile b/FreeFileSync/Source/Makefile
index b22960d..b9d06c8 100644
--- a/FreeFileSync/Source/Makefile
+++ b/FreeFileSync/Source/Makefile
@@ -24,6 +24,7 @@ CXXFLAGS  += `$(PKG_CONFIG) --cflags libssh2`
 LDFLAGS += `$(PKG_CONFIG) --libs   libssh2`
 
 CXXFLAGS  += `$(PKG_CONFIG) --cflags gtk+-3.0`
+LDFLAGS += `$(PKG_CONFIG) --libs gtk+-3.0`
 #treat as system headers so that warnings are hidden:
 CXXFLAGS += -isystem/usr/include/gtk-3.0
 
diff --git a/FreeFileSync/Source/RealTimeSync/Makefile b/FreeFileSync/Source/RealTimeSync/Makefile
index c93d240..6c49714 100644
--- a/FreeFileSync/Source/RealTimeSync/Makefile
+++ b/FreeFileSync/Source/RealTimeSync/Makefile
@@ -12,6 +12,7 @@ LDFLAGS += `$(WX_CONFIG) --libs std, aui, richtext --debug=no` -pthread
 
 
 CXXFLAGS += `$(PKG_CONFIG) --cflags gtk+-3.0`
+LDFLAGS += `$(PKG_CONFIG) --libs gtk+-3.0`
 #treat as system headers so that warnings are hidden:
 CXXFLAGS += -isystem/usr/include/gtk-3.0
 
